semantic-python/src/Language/Python/Core.hs

```diff
@@ -16,13 +18,35 @@ import qualified TreeSitter.Python.AST as Py
 class Compile py where
   -- FIXME: we should really try not to fail
   compile :: (Member Core sig, Carrier sig t, Foldable t, MonadFail m) => py -> m (t Name)
+
   default compile :: (MonadFail m, Show py) => py -> m (t Name)
   compile = defaultCompile
 
+  compileCC :: (Member Core sig, Carrier sig t, Foldable t, MonadFail m) => py -> m (t Name) -> m (t Name)
```

Comment:
This is like `compile`, but it takes a continuation—the operation to sequence next—as an argument. This means that we can encode the compilation of return statements, no matter if they’re early or terminal, by using `compileCC` and explicitly ignoring the continuation